src/shared/install.c

```diff
@@ -68,15 +68,15 @@ typedef struct {
         size_t n_rules;
 } Presets;
 
-static inline bool unit_file_install_info_has_rules(UnitFileInstallInfo *i) {
+static inline bool unit_file_install_info_has_rules(const UnitFileInstallInfo *i) {
```

Comment:
adding const like in this commit is fine otoh, as here the data type is API and we hence know what it contains and what not.